parser/lex.go

```diff
@@ -0,0 +1,605 @@
+// Based on design first introduced in: http://blog.golang.org/two-go-talks-lexical-scanning-in-go-and
+// Portions copied and modified from: https://github.com/golang/go/blob/master/src/text/template/parse/lex.go
+
+package parser
```

Comment:
I think you have to remove L7 to make it work. `golint` enforces that it should basically look exactly like this:

```go
// Copyright ...

// Package parser ...
package parser
```

Check out https://github.com/jzelinskie/conventions for a quick read on Go library conventions.